scripts/register_sam.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import os
+
+from google.auth.transport.requests import (
+    Request,
+)
+from google.oauth2 import (
+    service_account,
+)
+
+from azul import (
+    config,
+    require,
+)
+from azul.dss import (
+    shared_credentials,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+import azul.tdr
+
+log = logging.getLogger(__name__)
+
+
+class SamRegistration:
+
+    def __init__(self):
+        self.http = azul.tdr.http()
+        with shared_credentials():
+            credentials = service_account.Credentials.from_service_account_file(
+                os.environ['GOOGLE_APPLICATION_CREDENTIALS']
+            )
+        self.credentials: service_account.Credentials = credentials
+
+    def register(self) -> None:
+        token = self.get_access_token()
+        response = self.http.request('POST',
+                                     f'{config.sam_service_url}/register/user/v1',
+                                     body='',
+                                     headers={'Authorization': f'Bearer {token}'})
+
+        if response.status == 201:
+            log.info('Google service account successfully registered with SAM.')
+        elif response.status == 409:
+            log.info('Google service account previously registered with SAM.')
+        else:
+            raise RuntimeError(f'Unexpected response during SAM registration: {response.data}')
+
+    def get_access_token(self) -> str:
+        credentials = self.credentials.with_scopes(['email', 'openid'])
+        credentials.refresh(Request())
+        return credentials.token
+
+    def test_access(self):
+        response = self.http.request('GET',
```

Comment:
Why isn't this in the TDR client?

Reply:
Didn't occur to me. Moved and refactored a bit so it's purpose is more clear in the new context.